[Typespec Migration] servicefabricmanagedclusters

iliu816 opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [x] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

iliu816 committed:
initial conversion

iliu816 committed:
sorted swaggers

iliu816 committed:
initial generation

iliu816 committed:
ManagedCluster supression

iliu816 committed:
managedvmsize, managedclustercodeversionresult, applicationtype

iliu816 committed:
applicationTypeVersions supressions

iliu816 committed:
application supression

iliu816 committed:
main.tsp: bumping commontype version and adding service operations interface

iliu816 committed:
nodetype supressions

iliu816 committed:
readd  @parameterVisibility to ApplicationTypeResource to override patch operations defaulting to considering parameters as optional

iliu816 committed:
serviceresource supression

iliu816 committed:
operationresult, operationstatus supression

iliu816 committed:
first pass cleaning up naming, model links, supressions in models.tsp

iliu816 committed:
linking client.tsp, changing names of request payload from "properties" to "parameters"

iliu816 committed:
removing non-common-type resource models, confirming properties of resource models match common-types

iliu816 committed:
cleanup remaining linter warnings

iliu816 committed:
generated updates to examples

iliu816 committed:
XMS_EXAMPLE_NOTFOUND_ERROR

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Microsoft.ServiceFabric](https://apiview.dev/Assemblies/Review/85d7c1900e0442b887c8924a6b580dbe?revisionId=9862192c3cdc4b0989bb55ea448ec42d)

iliu816 committed:
add emitter for sdks

iliu816 committed:
readd example files for old swagger to reduce noise from errors

iliu816 committed:
added service-dir default, edits to java and go emitter options

iliu816 committed:
RESPONSE_NOT_IN_SPEC, RESPONSE_NOT_IN_EXAMPLE,: removed references to 200 responses for lro

iliu816 committed:
remove previous swagger version, readme linter warnings

iliu816 committed:
move from common-type v5 to v3 due to armid validation failure

iliu816 committed:
change eTag to etag

iliu816 committed:
add url to list operation examples

iliu816 committed:
removing requirement on location property for cluster, nodetype, and application models

iliu816 committed:
better aligning ManagedClusterCodeVersionResult definition with existing swagger

iliu816 committed:
specifying x-ms-parameter-location for operationId parameter

iliu816 committed:
better aligning ManagedVMSize with existing swagger

iliu816 committed:
spellcheck omit tsp files, revert managedcluster to trackedresource

iliu816 committed:
prettier check on example files

iliu816 committed:
spellcheck omission for tsp files

iliu816 committed:
remove duplicate tags property from managedcluster

iliu816 committed:
fix filepath for spellcheck omission for tsp files

iliu816 committed:
lintdiff for readme.md

iliu816 commented:
@microsoft-github-policy-service agree company="Microsoft"

iliu816 commented:
/azp run

iliu816 committed:
Lintdiff: SecurityDefinitionsStructure. resource managers not expected to use @useauth

iliu816 committed:
LintDiff: DescriptionMustNotBeNodeName

iliu816 committed:
parent-child relation between ManagedCluster and application models

iliu816 committed:
DescriptionMustNotBeNodeName: SingletonPartitionScheme

iliu816 committed:
clean up <*> pattern from docstrings, breaking client generation from tsp files, assumed docstrings converted to xml or something

iliu816 committed:
attempt formatting HealthCheckWaitDuration and HealthCheckWaitDuration in readme.md

iliu816 committed:
remove typespec emitters, brownfield service

iliu816 committed:
remove duration directives from readme.md

iliu816 committed:
mark durations

iliu816 committed:
add java and go to sdk in readme.md

iliu816 committed:
linux is not supported managedcodeversion environment in existing sfmc swagger

iliu816 committed:
emit final-state-schema for startFaultSimulation and stopFaultSimulation in api-spec

iliu816 committed:
breaking change warnings from go sdk generation

iliu816 committed:
breaking change warnings from java sdk generation:
* azureasyncoperation headers
* type of vmssextensions settings and protected settings

iliu816 committed:
FaultSimulation*Parameters -> FaultSimluation*Content for dotnet SDK generation

iliu816 committed:
breaking change warnings from python sdk generation:
* default values changed

iliu816 committed:
breaking change warnings from dotnet sdk generation:
* unflattening service properties models
* representing all rollingUpgradeMonitoringPolicy propreties as timespans
* ManagedCluster sku marked as required
* ManagedClusterVersion supportExpiryUtc marked as date-time

iliu816 committed:
breaking changes warnings from dotnet sdk generation:
* servicePlacementTimeLimit should be a timespan

iliu816 committed:
readd non-dotnet emitters to validate sdk generation

mikeharder committed:
Merge branch 'main' into user/iliu/2024-11-01-preview-migrate

iliu816 committed:
breaking changes in go sdk from tspconfig:
* clients removed; reorg'd operations under new interfaces and tags to try fix
* fixed tagging for fault simulation operations

iliu816 committed:
tsp formatting fix

iliu816 committed:
breaking changes in go sdk from tspconfig:
* attempt specifying client name

iliu816 committed:
revert

iliu816 committed:
breaking changes in go sdk from tspconfig:
* attempt specifying client name

iliu816 committed:
add back-compat to main.tsp

iliu816 committed:
typespec validation: emitter configs, formatting

iliu816 committed:
Merge branch 'main' into user/iliu/2024-11-01-preview-migrate

iliu816 committed:
typespec version upgrade: fixing warnings

iliu816 committed:
litndiff:
* supressions
* removed requirement for nodetypesku.capacity
* changed duplicate example names

iliu816 committed:
breaking changes in go sdk:
* specifying format for lro createOrUpdate responses
* fixing operation names for ManagedAzResiliencyStatusClient and ManagedMaintenanceWindowStatusClient get()

other:
* spellcheck on armid -> arm id

iliu816 committed:
double-checking required properties for models, readd capacity as required for nodetypesku

IpConfigurationPublicIPAddressConfiguration did not exist in previous swagger, updated to PublicIPAddressConfiguration

added more justification to suppressions

iliu816 committed:
spellcheck: provisiong -> provisioning

iliu816 committed:
lintdiff: RequiredPropertiesMissingInResourceModel,  PatchBodyParametersSchema

breaking change:
revert to IPConfigurationPublicIPAddressConfiguration

a-santamaria commented on specification/servicefabricmanagedclusters/ServiceFabricManagedClusters.Management/models.tsp:
```diff
@@ -0,0 +1,3702 @@
+/**
+ * Describes the managed cluster resource properties.
+ */
+model ManagedClusterProperties {
```
do we need to have all the models in the same tsp or in the future we can have 1 common, 1 for cluster, 1 for node type, etc ?

iliu816 replied:
We can move them. I think they get left in the models.tsp files by services because that's where the converter puts them.

iliu816 commented on specification/servicefabricmanagedclusters/ServiceFabricManagedClusters.Management/legacy.tsp:
```diff
@@ -0,0 +1,323 @@
+using TypeSpec.Rest;
+
+namespace Azure.ResourceManager.Legacy {
+  interface LegacyOperations<
```
This file was created by the converter tool, I believe as a way to represent a previous standard for operations. Unfortunately, this structure didn't fit our non-standard operation spec either, so we don't actually use this file. I will remove it.

a-santamaria commented on specification/servicefabricmanagedclusters/ServiceFabricManagedClusters.Management/back-compat.tsp:
```diff
@@ -0,0 +1,10 @@
+using Azure.ClientGenerator.Core;
+using Microsoft.ServiceFabric;
+
+@@clientName(ManagedMaintenanceWindowStatusContent,
```
is this to make clients use the same names they had before not to create breaking changes?

iliu816 replied:
Yup!

iliu816 committed:
* remove unused legacy.tsp added by converter tool
* rename azresiliency and maintencewindow clients in java sdk generation

iliu816 committed:
sorted model types

iliu816 committed:
align PublicIPAddressConfiguration with existing spec

iliu816 committed:
dotnet: add backcompat model name for PublicIPAddressConfiguration

iliu816 committed:
tsp-client compare: minValues, defaults

iliu816 committed:
tsp-client compare: fix for x-ms-identifiers in FaultSimulationListResult

iliu816 committed:
* IPConfigurationPublicIPAddressConfiguration for go, java, javascript, and python
* remove above langauge sdks from automated swagger-based generation of sdk

iliu816 committed:
* added Basic Information section to readme.md
* removed readme.insert_lanaguge.md used in autorest gen
* tried moving options from readme.python.me to tspconfig
* renamed SF client name for python

iliu816 committed:
*python: remove no-namespace-folders option

iliu816 committed:
dotnet: rename-mapping

iliu816 committed:
dotnet: remove autorest based generation to reduce noise from generation errors

iliu816 committed:
*VMSize and ManagedClusterCodeVersions are not armresources, removed decorators.

* dotnet: added some rename mapping for models and operation ids

iliu816 commented:
/azp run

iliu816 committed:
Merge branch 'main' into user/iliu/2024-11-01-preview-migrate

iliu816 committed:
* Add @OpenAPI.extension("x-ms-azure-resource", true) to SubResource model for swagger diff
